django/db/models/indexes.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import unicode_literals
+
+import hashlib
+
+from django.utils.encoding import force_bytes
+
+__all__ = ['Index']
+
+# Max length restriction on names imposed by Oracle
+MAX_NAME_LENGTH = 30
```

Comment:
We can't really make this backend dependent if we want to store the index name as part of migrations. Migrations are potentially run on any db backend and should work on any.